core/shared/src/main/scala/zio/ZIOAppDefault.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2021 John A. De Goes and the ZIO Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio
+
+/**
+ * The entry point for a ZIO application.
+ *
+ * {{{
+ * import zio.ZIOApp
+ * import zio.Console._
+ *
+ * object MyApp extends ZIOAppDefault {
+ *
+ *   def run =
+ *     for {
+ *       _ <- printLine("Hello! What is your name?")
+ *       n <- readLine
+ *       _ <- printLine("Hello, " + n + ", good to meet you!")
+ *     } yield ()
+ * }
+ * }}}
+ */
+trait ZIOAppDefault extends ZIOApp {
+
+  type Environment = ZEnv
+
+  val layer: ZLayer[ZEnv with Has[ZIOAppArgs], Any, ZEnv] = ZLayer.environment
+
+  val tag: Tag[ZEnv] = Tag[ZEnv]
+}
+
+object ZIOAppDefault {
+
+  /**
+   * Creates a [[ZIOAppDefault]] from an effect, which can consume the
+   * arguments of the program, as well as a hook into the ZIO runtime
+   * configuration.
+   */
+  def apply(
+    run0: ZIO[ZEnv with Has[ZIOAppArgs], Any, Any],
+    hook0: RuntimeConfigAspect
+  ): ZIOAppDefault =
+    new ZIOAppDefault {
+      override def hook = hook0
+      def run           = run0
+    }
+
+  /**
+   * Creates a [[ZIOAppDefault]] from an effect, using the unmodified default
+   * runtime's configuration.
+   */
+  def fromEffect(run0: ZIO[ZEnv with Has[ZIOAppArgs], Any, Any]): ZIOAppDefault =
```

Comment:
We should probably be calling these `fromZIO` this point.

Reply:
Yes I noticed that too! Will change!